src/main/scala/dxWDL/compiler/GenerateIRTask.scala

```diff
@@ -77,11 +77,34 @@ case class GenerateIRTask(verbose: Verbose,
       Some(obj.flatMap {
         case (IR.PARAM_META_HELP, MetaValueElementString(text)) => Some(IR.IOAttrHelp(text))
         case (IR.PARAM_META_PATTERNS, MetaValueElementArray(array)) =>
-          Some(IR.IOAttrPatterns(array.flatMap {
+          Some(IR.IOAttrPatterns(IR.PatternsReprArray(array.flatMap {
             case MetaValueElementString(pattern) => Some(pattern)
             case _                               => None
-
-          }.toVector))
+          }.toVector)))
+        case (IR.PARAM_META_PATTERNS, MetaValueElementObject(obj)) =>
+          val name = obj.get("name") match {
```

Comment:
Can you factor out this section into a separate private method? Also, describe that it is trying to parse the `tag, class, patterns` attributes. An example would be helpful: 
```
         patterns: {
              name: ["*.txt", "*.tsv"],
              class: "file",
              tag: ["foo", "bar"]
          }
```

Reply:
I added a better example in the docs for IOAttrPatterns